servicetalk-encoding-api/src/main/java/io/servicetalk/encoding/api/ContentCodecBuilder.java

```diff
@@ -29,7 +29,7 @@
      * @return {@code this}
      * @see <a href="https://en.wikipedia.org/wiki/Zip_bomb">Zip Bomb</a>
      */
-    ContentCodecBuilder setMaxAllowedPayloadSize(int maxAllowedPayloadSize);
+    ContentCodecBuilder maxAllowedPayloadSize(int maxAllowedPayloadSize);
```

Comment:
This is an API breaking change, but I am not anticipating anyone using this. If any objections, we can gracefully deprecate it.

Reply:
On a similar note, this API has some mismatches with the Netty's meaning.
For Netty max translates to `max-allocation`, and its tracked as the internal buffer used to decode _one_ message. In our current servicetalk-encoding-api based solution, that was used to actually track the max payload size (ie. accumulated https://github.com/apple/servicetalk/blob/main/servicetalk-encoding-api/src/main/java/io/servicetalk/encoding/api/AbstractZipContentCodec.java#L382).
Regardless of the API mismatch, which we can address by deprecating the old flavor, we should decide if Netty's approach is enough for our needs. From my perspective, I think we are fine, because a single chunk can't be larger than `max-allocation`, and the number of chunks can be controlled by a subscription's demand.